### PR TITLE
Feature/go 2473

### PR DESCRIPTION
**Pull request by cingalia**

### CODE REVIEW

File: .project [hunk 1]
```diff
@@ -14,4 +14,15 @@
 	<natures>
 		<nature>org.eclipse.m2e.core.maven2Nature</nature>
 	</natures>
+	<filteredResources>
```
Comment (kwburnett) [inline .project:17]: There have been quite a few PRs with this all in it. Can you separate it into it's own PR that we can merge so the other PRs don't have this in it (that doesn't pertain to the changes you're making)?

File: idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java [hunk 2]
```diff
@@ -126,9 +127,22 @@ public static String getLastVisitDate(MBPartner_BH patient) {
 
 	@Override
 	public Visit processEntity(String uuid, String docAction) throws Exception {
-		// We need to do something special for completing a sales order - do it
-		// asynchronously
-		if (StringUtil.isNullOrEmpty(docAction) || !docAction.equalsIgnoreCase(DocAction.ACTION_Complete)) {
+		int clientId = Env.getAD_Client_ID(Env.getCtx());
+		String clientIdsForSynchronousProcessingString =
```
Comment (kwburnett) [inline idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java]: I was thinking about this in the morning and, in alignment with something @andrewmoko had said previously, can we make it check UUIDs so that we won't have collisions on our other servers?

File: idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java [hunk 3]
```diff
@@ -147,8 +161,22 @@ public Visit processEntity(String uuid, String docAction) throws Exception {
 					newPayment.saveEx();
 				}
 				visit.setPayments(paymentDBService.getPaymentsByOrderId(visit.getId()));
-			}
+			} 
 			return visit;
+		} else if (docAction.equalsIgnoreCase(DocAction.ACTION_Complete)) {
```
Comment (kwburnett) [inline idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java]: What are these changes for?

File: package-lock.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (kwburnett) [inline package-lock.json]: You can omit this file from being committed.

**Review by kwburnett — CHANGES_REQUESTED** [review]

File: idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java (hunk 3 above)
Comment (cingalia) [reply idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java]: because it was inside reversal

File: idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java (hunk 2 above)
Comment (cingalia) [reply idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java]: What do you mean like check?

File: idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java (hunk 3 above)
Comment (cingalia) [reply idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java]: And since it's not reversal it will pass without processing

File: idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java (hunk 2 above)
Comment (kwburnett) [reply idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java]: Instead of checking against client IDs, can we make it check against client UUIDs?

File: idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java (hunk 3 above)
Comment (kwburnett) [reply idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java]: Gotcha. Then this can move to line 164 since we want it executing in the block above.

File: idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java [hunk 5]
```diff
@@ -126,9 +127,22 @@ public static String getLastVisitDate(MBPartner_BH patient) {
...
+				clientIdsForSynchronousProcessing = Arrays.stream(clientIdsForSynchronousProcessingString.split(","))
+						.map(stringClientId -> Integer.parseInt(stringClientId.trim())).collect(Collectors.toList());
+			}
+		} catch (Exception exception) {
+			log.severe(exception.getMessage());
+		}
+		// We need to do something special for completing a sales order - do it asynchronously (except for the clients we
+		// want to do it synchronously for)
+		if (StringUtil.isNullOrEmpty(docAction) || (!docAction.equalsIgnoreCase(DocAction.ACTION_Complete) &&
+				!clientIdsForSynchronousProcessing.contains(clientId))) {
```
Comment (kwburnett) [inline idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java]: We want this code block to execute if the client ID is specified for synchronous processing, so maybe:
```
if (StringUtil.isNullOrEmpty(docAction) || !docAction.equalsIgnoreCase(DocAction.ACTION_Complete) ||
				clientIdsForSynchronousProcessing.contains(clientId)))
```

**Review by kwburnett — CHANGES_REQUESTED** [review]

File: idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java [hunk 6]
```diff
@@ -38,6 +39,8 @@
 import org.bandahealth.idempiere.rest.utils.QueryUtil;
 import org.bandahealth.idempiere.rest.utils.SqlUtil;
 import org.bandahealth.idempiere.rest.utils.StringUtil;
+import org.compiere.model.I_AD_Client;
```
Comment (kwburnett) [inline idemp-rest/WEB-INF/src/org/bandahealth/idempiere/rest/service/db/VisitDBService.java]:
```suggestion
```

File: .gitignore [hunk 7]
```diff
@@ -184,4 +184,5 @@ modules.xml
 # https://plugins.jetbrains.com/plugin/12206-codestream
 .idea/codestream.xml
 
-# End of https://www.toptal.com/developers/gitignore/api/intellij,eclipse
\ No newline at end of file
+# End of https://www.toptal.com/developers/gitignore/api/intellij,eclipse
+/package-lock.json
```
Comment (kwburnett) [inline .gitignore:188]:
```suggestion
```

**Review by kwburnett — CHANGES_REQUESTED** [review]

**Review by kwburnett — DISMISSED** [review]
I sent the info on Discord, but there are still a few comments to be addressed.

**Review by andrewmoko — APPROVED** [review]
